src/composables/widgets/useComboWidget.ts

```diff
@@ -0,0 +1,414 @@
+  '.jpeg',
+  '.png',
+  '.gif',
+  '.webp',
+  '.bmp',
+  '.tiff',
+  '.svg',
+  // Video formats
+  '.mp4',
+  '.avi',
+  '.mov',
+  '.webm',
+  '.mkv',
+  '.flv',
+  '.wmv',
+  // Audio formats
+  '.mp3',
+  '.wav',
+  '.flac',
+  '.aac',
+  '.ogg',
+  '.m4a',
+  '.wma'
+]
+
+/**
+ * Check if options contain filename-like values (UUIDs or legacy hashes)
+ */
+function hasFilenameOptions(options: any[]): boolean {
+  return options.some((opt: any) => {
+    if (typeof opt !== 'string') return false
+
+    // Check for UUID format (new system)
+    const isUUID =
+      /^[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}$/i.test(
+        opt
+      )
+
+    // Check for common file extensions (legacy)
+    const hasExtension = FILE_EXTENSIONS.some((ext) =>
+      opt.toLowerCase().endsWith(ext)
+    )
+
+    // Check for hash-like filenames (legacy ComfyUI hashed files)
+    const isHashLike = /^[a-f0-9]{8,}\./i.test(opt)
+
+    return isUUID || hasExtension || isHashLike
+  })
+}
+
+/**
+ * Apply filename mapping to a widget using a simplified approach
+ */
+function applyFilenameMappingToWidget(
+  widget: IComboWidget,
+  node: LGraphNode,
+  _inputSpec: ComboInputSpec
+) {
+  // Validate widget exists
+  if (!widget) {
+    return
+  }
+
+  // Simple approach: just override _displayValue for text display
+  // Leave all widget functionality intact
+
+  // Cast to extended interface for type safety
+  const mappingWidget = widget as IFilenameMappingWidget
+
+  // Override serializeValue to ensure asset ID is used for API
+  mappingWidget.serializeValue = function () {
+    // Always return the actual widget value (asset ID) for serialization
+    return mappingWidget.value
+  }
+
+  // Override _displayValue to show human-readable names
+  try {
+    Object.defineProperty(mappingWidget, '_displayValue', {
+      get() {
+        if (mappingWidget.computedDisabled) return ''
+
+        // Get current asset ID value
+        const assetId = mappingWidget.value
+        if (typeof assetId !== 'string') return String(assetId)
+
+        // Try to get human-readable name from cache (deduplicated for display)
+        const mapping = fileNameMappingService.getCachedMapping('input', true)
+        const humanName = mapping[assetId]
+
+        // Return human name for display, fallback to asset ID
+        return humanName || assetId
+      },
+      configurable: true
+    })
+  } catch (error) {
+    // Property might be non-configurable, continue without override
+  }
+
+  // Also override the options.values to show human names in dropdown
+  const originalOptions = mappingWidget.options as any
+
+  // Store original values array - maintain the same array reference
+  const rawValues = Array.isArray(originalOptions.values)
+    ? originalOptions.values
+    : []
+
+  // Create a computed property that returns mapped values
+  if (mappingWidget.options) {
+    try {
+      Object.defineProperty(mappingWidget.options, 'values', {
```

Comment:
add a new widget and drop the old widget.

mappedInputWidget (think of better name). don't abuse the combo widget